plugins/aws/autoscaling/asgMissingELB.js

```diff
@@ -0,0 +1,105 @@
+    link: 'https://docs.aws.amazon.com/autoscaling/ec2/userguide/attach-load-balancer-asg.html',
+    recommended_action: 'Ensure that the AutoScaling group load balancer has not been deleted. If so, remove it from the ASG.',
+    apis: ['AutoScaling:describeAutoScalingGroups', 'ELB:describeLoadBalancers', 'ELBv2:describeLoadBalancers'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+        var elbNames = [];
+
+        async.each(regions.autoscaling, function(region, rcb){
+            var autoScalingGroups = helpers.addSource(cache, source,
+                ['autoscaling', 'describeAutoScalingGroups', region]);
+
+            var elasticLoadBalancers = helpers.addSource(cache, source,
+                ['elb', 'describeLoadBalancers', region]);
+
+            var elasticLoadBalancersV2 = helpers.addSource(cache, source,
+                ['elbv2', 'describeLoadBalancers', region]);
+
+            if (!autoScalingGroups || !elasticLoadBalancers || !elasticLoadBalancersV2) return rcb();
+
+            if (autoScalingGroups.err || !autoScalingGroups.data) {
+                helpers.addResult(results, 3, 'Unable to query for AutoScaling groups: ' +  helpers.addError(autoScalingGroups), region);
+                return rcb();
+            }
+
+            if (elasticLoadBalancers.err || !elasticLoadBalancers.data) {
+                helpers.addResult(results, 3, 'Unable to query for Classic load balancers: ' +  helpers.addError(elasticLoadBalancers), region);
+                return rcb();
+            }
+
+            if (elasticLoadBalancersV2.err || !elasticLoadBalancersV2.data) {
+                helpers.addResult(results, 3, 'Unable to query for Application/Network load balancers: ' +  helpers.addError(elasticLoadBalancersV2), region);
+                return rcb();
+            }
+            
+            if (!autoScalingGroups.data.length) {
+                helpers.addResult(results, 0, 'No AutoScaling group found', region);
+                return rcb();
+            }
+            
+            if (elasticLoadBalancers.data.length) {
+                elasticLoadBalancers.data.forEach(function(elb) {
+                    if(elb.LoadBalancerName) {
+                        elbNames.push(elb.LoadBalancerName);
+                    }
+                });
+            }
+
+            if (elasticLoadBalancersV2.data.length) {
+                elasticLoadBalancersV2.data.forEach(function(elbv2) {
+                    if(elbv2.LoadBalancerName) {
+                        elbNames.push(elbv2.LoadBalancerName);
+                    }
+                });
+            }
+
+            autoScalingGroups.data.forEach(function(asg){
+                var resource = asg.AutoScalingGroupARN;
+                var inactiveElbs = [];
+                if(asg.HealthCheckType == 'ELB') {
```

Comment:
check if `asg.HealthCheckType` exists before doing comparison

Reply:
@giorod3 _asg.HealthCheckType_  exists always. it can either be **EC2** or **ELB**. By default its set to EC2.